bspump/aio/sink.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import asyncio
+
+from ..abc.sink import Sink
+
+#
+
+L = logging.getLogger(__name__)
+
+#
+
+
+class AsyncSink(Sink):
+	"""
+	Inherit from the sink and override the `consume` coroutine:
+
+	class MySink(bspump.aio.AsyncSink):
+
+		async def consume(self, context, event):
+			...
+	"""
+
+	ConfigDefaults = {
+		"output_queue_max_size": 100,
+	}
+
+
+	def __init__(self, app, pipeline, id=None, config=None):
+		super().__init__(app, pipeline, id, config)
+
+		self.Queue = asyncio.Queue()
+		self.QueueMaxSize = int(self.Config["output_queue_max_size"])
+		assert (self.QueueMaxSize >= 1)
+
+		self.Task = asyncio.ensure_future(
+			self._task()
+		)
+
+		self.Exiting = False
+
+		app.PubSub.subscribe("Application.exit!", self._on_exit)
```

Comment:
There should be automatic Task restart when exception is raised in `consume()` + good logging of that exception + throttling of the pipeline

Reply:
@ateska The exception handling is already there, there is a while loop in the task with try catch inside.